fe/fe-core/src/main/java/org/apache/doris/planner/external/IcebergScanProvider.java

```diff
@@ -102,6 +124,25 @@ public static void setIcebergParams(ExternalFileScanNode.ParamCreateContext cont
         context.params.setTableFormatParams(tableFormatFileDesc);
     }
 
+    private static void setPathSelectConjunct(TIcebergFileDesc fileDesc, IcebergSplit icebergSplit)
+                throws UserException {
+        DeleteFileTempTable deleteFileTable = icebergSplit.getDeleteFileTable();
+        DescriptorTable descTbl = icebergSplit.getAnalyzer().getDescTbl();
+        TableRef ref = new TableRef(deleteFileTable.getTableName(), null, null);
+        BaseTableRef tableRef = new BaseTableRef(ref, deleteFileTable, deleteFileTable.getTableName());
+        tableRef.analyze(icebergSplit.getAnalyzer());
+        fileDesc.setTupleId(tableRef.getDesc().getId().asInt());
+        descTbl.computeStatAndMemLayout();
+
+        SlotRef lhs = new SlotRef(deleteFileTable.getTableName(), DeleteFileTempTable.DATA_FILE_PATH);
+        lhs.analyze(icebergSplit.getAnalyzer());
```

Comment:
No need to analyze `lhs` separately.
You can just analyze `pathSelectConjunct`.

Reply:
because we should use  lhs.getDesc().setIsMaterialized(true);